src/table/components/head/MenuList/MenuGroup.tsx

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable import/no-cycle */
+import React, { useState, useRef } from 'react';
+import Typography from '@mui/material/Typography';
+import ArrowRight from '@qlik-trial/sprout/icons/react/ArrowRight';
+import { HeadCellMenuItem, MenuItemGroup } from '../../../types';
+import { StyledMenuItem, StyledListItemIcon, StyledMenuItemLabel } from '../styles';
+import RecursiveMenuList from './RecursiveMenuList';
+
+export const interceptClickOnMenuItems = (menuGroups: MenuItemGroup[], cache: SubMenusOpenStatusCache) => {
+  const result = menuGroups.map((grp) => {
+    return grp.map(({ onClick, ...restProps }) => ({
+      ...restProps,
+      ...(onClick
+        ? {
+            onClick: (evt: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+              // reset all opened submenu levels here!
+              Object.entries(cache).map(([, setter]) => setter(false));
+              onClick(evt);
+            },
+          }
+        : {}),
+    }));
+  });
+  return result;
+};
+
+type SubMenusOpenStatusCache = Record<string, React.Dispatch<React.SetStateAction<boolean>>>;
+const subMenusOpenStatusCache: SubMenusOpenStatusCache = {};
+
+const MenuGroupItems = ({ id, onClick, itemTitle, icon, enabled, subMenus }: HeadCellMenuItem) => {
+  const [openMenu, setOpenMenu] = useState(false);
+  const anchorRef = useRef<HTMLDivElement | null>(null);
+
+  const handleOnClick = (evt: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+    if (onClick) onClick(evt);
+    if (subMenus?.length) {
+      setOpenMenu(true);
+      subMenusOpenStatusCache[itemTitle] = setOpenMenu;
```

Comment:
Would `id` not be a better key here? Not that I think we will run into duplicate item titles

Reply:
No i think, because each group can have an item with id of 1, 2 ...

Reply:
No i think, because each group can have an item with id of 1, 2 and so on. The thing that makes the keys of that object unique is going to be the titles because we cannot have a menu with same titles but each group can have duplicate ids

Reply:
Actually I wonder if this will work because `subMenusOpenStatusCache` will be shared by all sn-table instances

Reply:
but everything happens in scope of one table instance (in sense client for example)

Reply:
What I mean is that, even if you had 5 sn-table instances on a sheet and open the submenu for all of them. `const subMenusOpenStatusCache: SubMenusOpenStatusCache = {};` is only called once. So they all share the same cache.

But since they all have the same key (Selections) and can never be opened at the same time. It should not be a real issue. At least not for this case.

Reply:
Yes, since if there is one dropdown open already, user cannot open another dropdown from other table in a sheet => it will not going to be the case then!

Reply:
> also each cache is specific for one table, so if we have 5 table instances there will be 5 cache objects for each

No I don't think so. The creation of the cache is only run once. So ALL instances will share the same cache?